lib/geocoder/lookups/azure.rb

```diff
@@ -0,0 +1,56 @@
+require 'geocoder/lookups/base'
+require 'geocoder/results/azure'
+
+module Geocoder::Lookup
+  class Azure < Base
+    def name
+      'Azure'
+    end
+
+    def required_api_key_parts
+      ['api_key']
+    end
+
+    def supported_protocols
+      [:https]
+    end
+
+    private
+
+    def base_query_url(query)
+      host = 'atlas.microsoft.com/search/address'
+
+      if query.reverse_geocode?
+        "#{protocol}://#{host}/reverse/json?"
+      else
+        "#{protocol}://#{host}/json?"
+      end
+    end
+
+    def query_url_params(query)
+      params = {
+        'api-version' => 1.0,
+        'language' => query.options[:language] || 'en',
+        'limit' => configuration.limit || 10,
+        'query' => query.sanitized_text,
+        'subscription-key' => configuration.api_key
+      }
+
+      params.merge!(super)
```

Comment:
For readability, I would remove the `!` from the method name, since `params` is discarded when the method exits (the in-place merging is irrelevant, we're just returning the result of the merge).